src/frontends/pytorch/src/transforms.cpp

```diff
@@ -0,0 +1,419 @@
+//
+
+#include "transforms.hpp"
+
+#include <iostream>
+#include <tuple>
+
+#include "openvino/frontend/pytorch/decoder.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+
+using ov::op::util::FrameworkNode;
+using ov::pass::MatcherPass;
+using ov::pass::pattern::any_input;
+using ov::pass::pattern::Matcher;
+using ov::pass::pattern::wrap_type;
+using std::make_shared;
+using std::shared_ptr;
+
+const type::List* is_list(const descriptor::Tensor& tensor) {
+    // TODO: Use special API to get custom type detalization
+    /*if (falsetensor.get_element_type() == element::custom) {
+        auto custom_type = tensor.get_custom_element_type();
+        if(custom_type.is<type::List>()) {
+            return &custom_type.as<type::List>();
+        }
+    }*/
+
+    return nullptr;
+}
+
+std::tuple<bool, Any> is_list_of_tensors(const descriptor::Tensor& tensor) {
+    // TODO: Use special API to get custom type detalization
+    /*if (auto list = is_list(tensor)) {
+        if(list->element_type.is<type::Tensor>()) {
+            return std::make_tuple(true, tensor.get_custom_element_type());  // UGLY: used custom type from the top
+    again
+        }
+    }
+
+    return std::make_tuple(false, Any());
+
+    if (tensor.get_element_type() != element::custom)
+        return std::make_tuple(false, Any());
+    Any custom_type = tensor.get_custom_element_type();*/
+    Any custom_type;
+    if (custom_type.empty()) {
+        return std::make_tuple(false, Any());
+    }
+
+    if (!custom_type.is<type::List>()) {
+        return std::make_tuple(false, custom_type);
+    }
+
+    Any element_type = custom_type.as<type::List>().element_type;
+
+    if (!element_type.is<type::Tensor>()) {
+        return std::make_tuple(false, custom_type);
+    }
+
+    return std::make_tuple(true, custom_type);
+}
+
+std::shared_ptr<FrameworkNode> make_list_pack(const OutputVector& inputs, Any output_type, const PartialShape& shape) {
+    auto list_pack = make_shared<FrameworkNode>(inputs, 1);  // 6 inputs -- 1 output
+    if (output_type.empty()) {
+        throw std::runtime_error("Attemt to call make_list_pack with empty output_type");
+    }
+    // TODO: Use special API to set custom type detalization
+    // list_pack->set_custom_output_type(0, output_type, shape);
+    ov::op::util::FrameworkNodeAttrs attrs;
+    attrs.set_type_name("PTFE::ListPack");
+    list_pack->set_attrs(attrs);
+    list_pack->validate_and_infer_types();
+    return list_pack;
+}
+
+std::shared_ptr<FrameworkNode> cast_internal_node(std::shared_ptr<Node> node, const std::string& type) {
+    auto fw_node = std::dynamic_pointer_cast<FrameworkNode>(node);
+    if (!fw_node) {
+        // std::cerr << "[ ERROR ] Incorrect matcher triggering\n";
```

Comment:
can you please remove this commented out code or use DEBUG routines?
The same comment for other lines